packages/stark-core/src/settings/services/settings.service.spec.ts

```diff
@@ -0,0 +1,172 @@
+			lastName: "Tenret",
+			language: "FR",
+			email: "tenretc@nbb.be",
+			referenceNumber: "1234"
+		};
+
+		(<Spy>mockStore.select).and.returnValue(of(mockUser));
+
+		settingsService = new SettingsServiceHelper(mockLogger, mockStore, appMetadata, appConfig);
+		// reset the calls counter because there is a log in the constructor
+		(<Spy>mockStore.dispatch).calls.reset();
+	});
+	describe("on initialization", () => {
+		it("preferredLanguage should be undefined", () => {
+			expect(settingsService.preferredLanguage).toBe(appConfig.defaultLanguage);
+		});
+	});
+
+	describe("initializeSettings", () => {
+		beforeEach(() => {
+			Object.defineProperty(navigator, "language", {
+				get: () => "fr-be"
+			});
+			browserLanguageCode = "fr";
+
+			spyOn(settingsService, "setPreferredLanguage").and.callFake((language: string) => {
+				settingsService.preferredLanguage = language;
+			});
+		});
+
+		it("should set user language as preferred language", () => {
+			settingsService.initializeSettings();
+			expect(settingsService.preferredLanguage).toEqual(<string>mockUser.language);
+		});
+
+		it("should set browser language as preferred language if user language is undefined OR null", () => {
+			/* tslint:disable:no-null-keyword */
+			mockUser.language = <any>null;
+			/* tslint:enable */
+			(<Spy>mockStore.select).and.returnValue(of(mockUser));
+			settingsService.initializeSettings();
+			expect(settingsService.preferredLanguage).toEqual(browserLanguageCode);
+
+			mockUser.language = undefined;
+			(<Spy>mockStore.select).and.returnValue(of(mockUser));
+			settingsService.initializeSettings();
+			expect(settingsService.preferredLanguage).toEqual(browserLanguageCode);
+		});
+
+		it("should set browser language as preferred language if user language is NOT in supported languages", () => {
+			mockUser.language = "NL";
+			(<Spy>mockStore.select).and.returnValue(of(mockUser));
+			settingsService.initializeSettings();
+			expect(settingsService.preferredLanguage).toEqual(browserLanguageCode);
+		});
+
+		it("should set defaultLanguage if browserLanguage not supported", () => {
+			appMetadata.supportedLanguages = [];
+			settingsService.initializeSettings();
+			expect(settingsService.preferredLanguage).toEqual(appConfig.defaultLanguage);
+		});
+	});
+
+	describe("getPreferredLanguage", () => {
+		it("should return the preferred language", () => {
+			const getPreferredLanguage: string = settingsService.getPreferredLanguage();
+			expect(getPreferredLanguage).toEqual(appConfig.defaultLanguage);
+		});
+	});
+
+	describe("setPreferredLanguage", () => {
+		it("should change the language successfully and dispatch the SUCCESS action", () => {
+			settingsService.setPreferredLanguage("NL");
+
+			expect(mockStore.dispatch).toHaveBeenCalledTimes(1);
+			expect((<Spy>mockStore.dispatch).calls.argsFor(0)[0]).toEqual(new SetPreferredLanguage("NL"));
+		});
+	});
+});
+
+class SettingsServiceHelper extends StarkSettingsServiceImpl {
```

Comment:
Why this Helper class? If you don't have any custom logic in here, there is no point to have this class and you can directly use the StarkSettingsServiceImpl